jmxtrans-output/jmxtrans-output-elastic/src/main/java/com/googlecode/jmxtrans/model/output/elastic/ElasticWriter.java

```diff
@@ -91,19 +100,24 @@ public ElasticWriter(
 						rootPrefix,
 						(String) getSettings().get("rootPrefix"),
 						DEFAULT_ROOT_PREFIX);
+		this.readTimeout = firstNonNull(connTimeout, (Integer) getSettings().get("connTimeout"), ELASTIC_CONNECTION_TIMEOUT);
+		this.connTimeout = firstNonNull(readTimeout, (Integer) getSettings().get("readTimeout"), ELASTIC_READ_TIMEOUT);
 
 		this.connectionUrl = connectionUrl;
+
+		this.username = username;
 		this.indexName = this.rootPrefix + "_jmx-entries";
-		this.jestClient = createJestClient(connectionUrl);
+		this.jestClient = createJestClient(connectionUrl, username, password);
 	}
 
-	private JestClient createJestClient(String connectionUrl) {
+	private JestClient createJestClient(String connectionUrl, String username, String password) {
 		log.info("Create a jest elastic search client for connection url [{}]", connectionUrl);
 		JestClientFactory factory = new JestClientFactory();
-		factory.setHttpClientConfig(
-				new HttpClientConfig.Builder(connectionUrl)
-						.multiThreaded(true)
-						.build());
+		HttpClientConfig.Builder httpClientBuilder  = new HttpClientConfig.Builder(connectionUrl).connTimeout(Integer.valueOf(connTimeout)).readTimeout(Integer.valueOf(readTimeout)).multiThreaded(true);
+		if( username != null && username.trim() != "") {
```

Comment:
Yeah, `StringUtils` would be even better. It seems I'm not fully awake yet... Thanks @lookfirst!